macOS/DuckDuckGo/MainWindow/MainWindow.swift

```diff
@@ -78,18 +79,11 @@ final class MainWindow: NSWindow {
         // The only reliable way to detect NSTextField is the first responder
         defer {
             // Send it after the first responder has been set on the super class so that window.firstResponder matches correctly
-            postFirstResponderNotification(with: responder)
+            NotificationCenter.default.post(name: MainWindow.firstResponderDidChangeNotification, object: self)
         }
-
         return super.makeFirstResponder(responder)
     }
 
-    override func becomeMain() {
-        super.becomeMain()
-
-        postFirstResponderNotification(with: firstResponder)
```

Comment:
we didn‘t bind the first responder notification to a concrete window object, that‘s why we‘ve had to post another notification on window activation; sending the notification with the window object should fix it